source/exe/main_common.cc

```diff
@@ -186,12 +186,15 @@ bool MainCommonBase::run() {
 
 void MainCommonBase::adminRequest(absl::string_view path_and_query, absl::string_view method,
                                   const AdminRequestFn& handler) {
+  ASSERT(server_->admin().has_value(), "adminRequest called with admin support compiled out");
```

Comment:
ok having this assert in in debug mode seems OK. But if we are not in debug mode should we return a 501 or something?

Seems like the fall-through case is going to jus not populate response_headers at all which might be confusing.

Alternatively -- I know you are not going to like this, but this is really a C++ API for folks that instantiate main_common themselves and can call methods on it. You could ifdef out the existence of this method entirely so calling it would be a build failure rather than a crash. But it's up to you.

Reply:
I don't love it, but I'll take it over a custom error path no one actually wants but I'd otherwise feel obliged to write tests for =P

Reply:
Great point!